app/models/has_corporate_information_pages.rb

```diff
@@ -4,7 +4,7 @@ module HasCorporateInformationPages
   included do
     has_many :corporate_information_pages, through: "edition_#{table_name}".to_sym, source: :edition, class_name: "CorporateInformationPage"
     before_destroy do |record|
-      record.corporate_information_pages.map { |cip| cip.document.destroy }
+      record.corporate_information_pages.map(&:document).uniq.each(&:destroy)
```

Comment:
We can't use `delete_all` as this won't fire callbacks on the `Document`. The alternative is `destroy_all` which only accepts a conditions hash and doesn't work on an AR scope.